### PR TITLE
Adds an in-game mentorhelp system as well as shuffles around "staffhelp" verbs

### PR DESCRIPTION
**Pull request by Shadowtail117**
## About the Pull Request

This PR adds a new system for mentors and mentorhelps, plus making some "I need help" verbs easier to access.

Front-end changes mostly coincide with back-end changes and most of the changes made are common sense/easily identifiable in the diffs. I am happy to elaborate if anything is unclear.

## Why It's Good For The Game

Mentors deserve usefulness in-game. Also, de-cluttering the verb panel is always good.

## Did you test it?

Yes, repeatedly during development and a final check of everything that I could reasonably test.

## Changelog

:cl:
admin: Mentors can now be specified in adminranks.txt with the "+MENTOR" permission, or with "+EVERYTHING". Staff ranks that have ONLY +MENTOR will not receive the default admin verbs or access to the MC panel, and cannot deadmin themselves.
rscadd: All players now have access to a "Mentorhelp" verb, which allows you to request help from an in-game mentor similar to an adminhelp.
rscadd: Mentors can respond to a player's mentorhelp by clicking on their name (players can reply to a mentor in the same way) and can also initiate the conversation by either right-clicking a mob or in the "Mentor PM" verb.
admin: Msay has received an overhaul and now stands for mentorsay, not moderatorsay. Mentors have access to it by default. Admins retain access.
tweak: All of the verbs listed above, as well as adminhelp, asay, admin PM, and staffwho are now in a new "Staffhelp" category for easier access.
admin: Admin memos have been moved to an "Admin Memos" verb, and a new "Mentor Memos" verb has been added with nearly-identical functionality. They both reside in the Admin verb category. Mentors cannot edit or delete mentor memos or see admin memos.
admin: Mentor action logs (mentorhelps and PMs) can be viewed in the Secrets panel.
admin: Two new config settings have been added, LOG_MENTOR and ENABLE_MEMOS. The former acts similarly to LOG_ADMIN, and the latter will enable/disable the use of the memo systems (though the verbs will still be visible).
admin: Admins can mute people from using mentorhelps and mentor PMs.
/:cl:

### CODE REVIEW

**Comment by Cupax3** [comment]
Squash please

**Review by Cupax3 — CHANGES_REQUESTED** [review]
Can you change the chat icon to "MSAY" by the by?

**Comment by Shadowtail117** [comment]
> Can you change the chat icon to "MSAY" by the by?

Addressed. Will squoosh when ready to merge.

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -52,6 +52,7 @@ var/list/admin_verbs_admin = list(
 	/client/proc/manage_silicon_laws,	// Allows viewing and editing silicon laws. ,
 	/client/proc/check_antagonists,
 	/client/proc/admin_memo,			//admin memo system. show/delete/write. +SERVER needed to delete admin memos of others,
+	/client/proc/mentor_memo,			//Ditto but for mentors
```
Comment (Cupax3) [inline code/modules/admin/admin_verbs.dm]: Append , to the comments in lists please, that should solve the check not clearing.

**Review by Cupax3 — CHANGES_REQUESTED** [review]
Try the suggested to resolve the check weirdness, otherwise LGTM

File: code/modules/admin/admin_verbs.dm (hunk 1 above)
Comment (Shadowtail117) [reply code/modules/admin/admin_verbs.dm]: I don't even know how or why it worked when it is a **COMMENT** but it did. Thanks.

Global var hash is still screaming at me which I'm not sure how to fix.

**Comment by Cupax3** [comment]
Apologies but this now conflicts with... #59 - can you rebase it to current dev and re-hash? Also gives an opportunity to squash, which would've been needed anyways. :D

**Comment by Shadowtail117** [comment]
Here goes nothing with the rebase please don't break everything

**Review by Cupax3 — APPROVED** [review]
LGTM

**Pull request merged**